pkgs/by-name/cy/cyrus-imapd/package.nix

```diff
@@ -0,0 +1,191 @@
+      cyrus_sasl.dev
+      icu
+      jansson
+      libbsd
+      libuuid
+      openssl
+      zlib
+      bison
+      libsrs2
+    ]
+    ++ lib.optional stdenv.isLinux libcap
+    ++ lib.optionals (withHttp || withCalalarmd || withJMAP) [
+      brotli.dev
+      libical.dev
+      libxml2.dev
+      nghttp2.dev
+      shapelib
+    ]
+    ++ lib.optionals withJMAP [
+      libchardet
+      wslay
+    ]
+    ++ lib.optional withXapian [
+      rsync
+      xapian
+    ]
+    ++ lib.optional withMySQL libmysqlclient
+    ++ lib.optional withPgSQL postgresql
+    ++ lib.optional withSQLite sqlite;
+
+  enableParallelBuilding = true;
+
+  # Copied from debian deb package
+  patches = [
+    ./patches/0002-Shutdown-and-close-sockets-cleanly.patch
+    ./patches/0003-Fix-syslog-prefix.patch
+    ./patches/0009-Normalize-the-authentication-ID.patch
+    ./patches/0018-increase-test-timeout.patch
+    ./patches/0019-propagate-XXFLAGS.patch
+  ];
+
+  postPatch = ''
+    patchShebangs cunit/*.pl
+    patchShebangs imap/promdatagen
+    patchShebangs tools/*
+
+    echo ${finalAttrs.version} > VERSION
+
+    substituteInPlace cunit/command.testc \
+      --replace-warn /usr/bin/touch ${coreutils}/bin/touch \
+      --replace-warn /bin/echo ${coreutils}/bin/echo \
+      --replace-warn /usr/bin/tr ${coreutils}/bin/tr \
+      --replace-warn /bin/sh ${stdenv.shell}
+
+    # libuuid on darwin isn't exist, try to build without it
+    substituteInPlace perl/imap/Makefile.PL.in \
+      --replace-fail  '"$LIB_SASL' '"-L${lib.getLib zlib}/lib ${
+        lib.optionalString (!stdenv.isDarwin) "-L${lib.getLib libuuid}/lib"
+      } -L${lib.getLib cyrus_sasl}/lib -L${lib.getLib pcre2}/lib -lpcre2-posix $LIB_SASL'
+    substituteInPlace perl/sieve/managesieve/Makefile.PL.in \
+      --replace-fail  '"$LIB_SASL' '"-L${lib.getLib zlib}/lib ${
+        lib.optionalString (!stdenv.isDarwin) "-L${lib.getLib libuuid}/lib"
+      } -L${lib.getLib cyrus_sasl}/lib $LIB_SASL'
+  '';
+
+  postInstall = ''
+    install -Dm755 ${./cyrus-cli} $out/bin/cyrus
+  '';
+
+  postFixup = ''
+    substituteInPlace $out/bin/cyrus \
+      --replace-fail PLACEHOLDER_BINPATH $out/bin
+    wrapProgram $out/bin/cyradm --set PERL5LIB $(find $out/lib/perl5 -type d | tr "\\n" ":")
+  '';
+
+  configureFlags = [
+    "--with-openssl=yes"
+    "--enable-autocreate"
+    "--enable-srs"
+    "--enable-idled"
+    "--enable-murder"
+    "--enable-backup"
+    "--enable-replication"
+    "--enable-unit-tests"
+    "--with-pidfile=/run/cyrus/master.pid"
+    "--with-zlib=${zlib}"
+    "--with-libcap=${libcap}"
+    (lib.enableFeature (withHttp || withCalalarmd || withJMAP) "http")
+    (lib.enableFeature withJMAP "jmap")
+    (lib.enableFeature withNNTP "nntp")
+    (lib.enableFeature withXapian "xapian")
+    (lib.enableFeature withCalalarmd "calalarmd")
+    (lib.withFeature withMySQL "mysql")
+    (lib.withFeature withPgSQL "pgsql")
+    (lib.withFeature withSQLite "sqlite")
+  ];
```

Comment:
Can't we use feature flags?

Reply:
Could you please give more details?

Reply:
See this: https://github.com/NixOS/nixpkgs/pull/307556#discussion_r1582430505

Reply:
Yes. The wording in my initial comment was not the best.